### DISCUSSION TITLE
Backport of sdbus-cpp to C++14

### DISCUSSION THREAD

**Discussion by beat-schaer** [Kistler-Group/sdbus-cpp#300] [Ideas]
I like sdbus-cpp very much and would like it to use for a customer. But unfortunately, it uses a target where the latest available compiler only speaks C++14.
Therefore, I consider a backport of sdbus-cpp to C++14. I have 2 questions concerning this:

1. Does anybody knows if this has already been done?
2. If not, is there any interest to adopt it into the main repository here as a backport branch?

**Comment by sangelovic** [comment]
@beat-schaer Hi, thanks for the message.

> Does anybody knows if this has already been done?

Originally, sdbus-c++ was written in C++14. But some years ago, the code was updated for C++17. (It was around this commit: 975f1bf07f2a0e55bc41e5af05bc2e54a8079d44.) So perhaps it make may it easier for you to look at how it was implemented before when porting back to C++14.

> If not, is there any interest to adopt it into the main repository here as a backport branch?

Yes, I think we could accept that. The only thing is additional maintenance it would incur... Would you be willing to then maintain this branch (synchronize with changes in main branch)? ;-)

**Reply by beat-schaer** [reply]
Yes I think I could do that...
